clientlibrary/worker/worker.go

```diff
@@ -204,19 +226,97 @@ func (w *Worker) initialize() error {
 	return nil
 }
 
+// fetchConsumerARNWithRetry tries to fetch consumer ARN. Retries 10 times with exponential backoff in case of an error
```

Comment:
It might be better to move fanout-specific consumer retrieval/creation logic into a different file. similar to the way how worker tests are done.
https://github.com/vmware/vmware-go-kcl/tree/master/test